openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,57 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    providers_data = None
+    cache_key = None
+    if credit_config.is_cache_enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
+        return providers_data
+
+    providers_data = []
```

Comment:
@rlucioni providers_data sets to None if cache value is not available.
```
providers_data = cache.get(cache_key)
```
That's why I initialized it here again.